### PR TITLE
Fix init debug msg

### PR DESCRIPTION
**Pull request by yfguo**
1. Added new debug messages for MPIR_Comm_commmit and MPIR_Intercomm_merge_impl.
2. Fixed the mismatched FUNC_ENTER/FUNC_EXIT in CH4
3. Added debug message to print the entire rank-address mapping for COMMs with less than 16 ranks.
4. Fixed uninitialized avtid in the rank-address translation.

### CODE REVIEW

**Comment by yfguo** [comment]
test:jenkins/ch3/all
test:jenkins/ch4/all

File: src/mpi/comm/commutil.c [hunk 1]
```diff
@@ -636,6 +636,7 @@ int MPIR_Comm_commit(MPIR_Comm * comm)
             comm->node_roots_comm->comm_kind = MPIR_COMM_KIND__INTRACOMM;
             comm->node_roots_comm->hierarchy_kind = MPIR_COMM_HIERARCHY_KIND__NODE_ROOTS;
             comm->node_roots_comm->local_comm = NULL;
+            MPL_DBG_MSG_D(MPIR_DBG_COMM, VERBOSE, "Create node_roots_comm=%p\n", comm->node_roots_comm);
```
Comment (raffenet) [inline src/mpi/comm/commutil.c]: I think you meant to use `MPL_DBG_MSG_P` here? The `_D` version is for integers.

File: src/mpid/ch4/src/ch4_init.h [hunk 2]
```diff
@@ -200,6 +200,15 @@ MPL_STATIC_INLINE_PREFIX int MPID_Init(int *argc,
...
     MPIDIU_avt_add_ref(0);
 
+#ifdef MPL_USE_DBG_LOGGING
+    int counter_;
+    if (size < 16) {
+        for (counter_ = 0; counter_ < size; ++counter_) {
+            MPIDIU_comm_rank_to_av(MPIR_Process.comm_world, counter_);
+        }
+    }
+#endif
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4_init.h]: The idea here is we basically do dummy lookups to force printing the translations?

Comment (yfguo) [inline src/mpid/ch4/src/ch4_init.h]: Yes, just to do dummy translation to print the addresses into the log. It gives us two things: 1) allowing us to confirm the correctness of rank-address(global rank) mapping; 2) address of addr, which is useful to cross check other part of the debug log.

File: src/mpi/comm/commutil.c (hunk 1 above)
Comment (yfguo) [inline src/mpi/comm/commutil.c]: You are right. Let me fix it.

File: src/mpid/ch4/src/ch4i_comm.h [hunk 3]
```diff
@@ -1153,6 +1153,22 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_comm_create_rank_map(MPIR_Comm * comm)
         MPIDI_COMM(comm, local_map).mode = MPIDI_RANK_MAP_NONE;
     }
 
+#ifdef MPL_USE_DBG_LOGGING
+    int rank_;
+    int avtid_, lpid_;
+    if (comm->remote_size < 16) {
+        for (rank_ = 0; rank_ < comm->remote_size; ++rank_) {
+            MPIDIU_comm_rank_to_pid(comm, rank_, &lpid, &avtid);
```
Comment (raffenet) [inline src/mpid/ch4/src/ch4i_comm.h]: did you mean to use the underscore versions of `lpid` and `avtid` variables here?

Comment (yfguo) [inline src/mpid/ch4/src/ch4i_comm.h]: Yes. This is a good catch. I have force pushed with the fix.

**Comment by yfguo** [comment]
test:jenkins/ch4/all

**Pull request merged**